libmamba/src/specs/version_spec.cpp

```diff
@@ -172,7 +290,13 @@ namespace mamba::specs
 
     auto operator==(const VersionPredicate& lhs, const VersionPredicate& rhs) -> bool
     {
-        return (lhs.m_operator == rhs.m_operator) && (lhs.m_version == rhs.m_version);
+        return (lhs.m_operator == rhs.m_operator)  //
+               && (lhs.m_version == rhs.m_version)
+               // In version_glob, the version is not understood purely as a version since ``*``
+               // has different meaning, as explicit trailing zeros.
+               // Versions should be made part of this variant internal and handled there.
```

Comment:
Is it worth referencing https://github.com/conda/ceps/pull/60 here?

```suggestion
               // Versions should be made part of this variant internal and handled there.
               // On the different meanings of ``*``, see this CEP: https://github.com/conda/ceps/pull/60
```